src/inference/src/dev/core_impl.cpp

```diff
@@ -800,11 +798,57 @@ ov::AnyMap ov::CoreImpl::get_supported_property(const std::string& full_device_n
     } catch (ov::Exception&) {
     }
 
+    // if device_name represents virtual device (software plugin), we need to add properties for
+    // its sub-devices, which are either virtual or hardware devices
+    if (is_virtual_device(device_name)) {
+        const auto priority_prop_name_it = options.find(priority_prop_name);
+        const std::string device_priorities_value =
+            priority_prop_name_it != options.end() ?
+                                                   // take user-set value
+                priority_prop_name_it->second.as<std::string>()
+                                                   :
+                                                   // or extract from the plugin if it's set globally
+                get_property(device_name, priority_prop_name, options).as<std::string>();
+
+        std::vector<std::string> sub_devices;
+        if (device_name == "HETERO")
+            sub_devices = ov::DeviceIDParser::get_hetero_devices(device_priorities_value);
+        else if (device_name == "AUTO" || device_name == "MULTI")
+            sub_devices = ov::DeviceIDParser::get_multi_devices(device_priorities_value);
+        else if (device_name == "BATCH")
+            sub_devices = {ov::DeviceIDParser::get_batch_device(device_priorities_value)};
+
+        /** go over flattened_config:
+         * - device::properties needs to be bypassed by as is to supported_config, because they
+         *   are actually handled / flattened in later calls
+         * - other properties need to be passed to get_supported_property call for sub devices
+         */
+        ov::AnyMap simple_properties;
+        for (const auto& property : flattened_config) {
+            const auto is_secondary_property = property.first.find(ov::device::properties.name()) != std::string::npos;
+            if (is_secondary_property) {
+                supported_config[property.first] = property.second;
+            } else {
+                simple_properties[property.first] = property.second;
+            }
+        }
+
+        // check simple / flat properties on priority devices (ov::device::priorities)
+        for (const auto& sub_device : sub_devices) {
+            auto sub_device_supported_properties = get_supported_property(sub_device, simple_properties);
+            for (auto&& kvp : sub_device_supported_properties) {
+                supported_config[kvp.first] = kvp.second;
+            }
+        }
+    }
+
+    // collect supported properties for device itself independently whether it's virtual or HW one
     for (auto&& kvp : flattened_config) {
         if (util::contains(supported_config_keys, kvp.first)) {
```

Comment:
ilya @ilya-lavrenov , for query network inferface, virtual plugins will also need to call this API to generate correct config list for each hardware plugin, but in this case, the core_level_properties will not be consumed by core, which will be propagated to hardware plugins and causing failures then.  probably, we should filter out the core_level_properties as well in query_model?

Reply:
but why core-level properties are passed to query_model at all?
If users did it, then they will have expected error when such properties come to HW plugins.

Currently AUTO plugin adds multiple properties, which are not set by users. We need to remove this logic in AUTO plugin, so AUTO should handle only its properties.

Reply:
for auto part, yes we are removing related.